pkg/radix/tree.go

```diff
@@ -396,174 +338,25 @@ func (t *Tree) Get(s string) (int, bool) {
 		}
 
 		// Consume the search prefix
-		if strings.HasPrefix(search, n.prefix) {
+		if bytes.HasPrefix(search, n.prefix) {
 			search = search[len(n.prefix):]
 		} else {
 			break
 		}
 	}
-	t.mu.RUnlock()
 	return 0, false
 }
 
-// LongestPrefix is like Get, but instead of an
-// exact match, it will return the longest prefix match.
-func (t *Tree) LongestPrefix(s string) (string, int, bool) {
-	t.mu.RLock()
-	defer t.mu.RUnlock()
-
-	var last *leafNode
-	n := t.root
-	search := s
-	for {
-		// Look for a leaf node
-		if n.isLeaf() {
-			last = n.leaf
-		}
-
-		// Check for key exhaution
-		if len(search) == 0 {
-			break
-		}
-
-		// Look for an edge
-		n = n.getEdge(search[0])
-		if n == nil {
-			break
-		}
-
-		// Consume the search prefix
-		if strings.HasPrefix(search, n.prefix) {
-			search = search[len(n.prefix):]
-		} else {
-			break
-		}
-	}
-	if last != nil {
-		return last.key, last.val, true
-	}
-	return "", 0, false
-}
-
-// Minimum is used to return the minimum value in the tree
-func (t *Tree) Minimum() (string, int, bool) {
```

Comment:
`Minimum` and maybe `Maximum` are needed in a branch I need to push up.  Can you add them back?

Reply:
No problem. Sorry for going overboard!